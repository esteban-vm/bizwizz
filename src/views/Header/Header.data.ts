import type { NavLinkProps } from '@/types'

const navLinks: NavLinkProps[] = [
  {
    id: 'e67da725-6069-4902-838f-2e54b680426e',
    text: 'Contact',
    to: '#contact',
  },
  {
    id: 'c054a01e-9103-4cd7-8a2f-c647d0d3e73b',
    text: 'Our services',
    to: '#services',
  },
  {
    id: 'fdd9fce5-598c-4029-8682-5296584eec75',
    text: 'Our Work',
    to: '#portfolio',
  },
]

export default navLinks
