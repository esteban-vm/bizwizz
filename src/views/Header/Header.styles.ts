import clsx from 'clsx'

const styles = {
  wrapper: clsx(
    'relative h-[100vh] w-full',
    'bg-black bg-[url(/images/header-img.jpg)] bg-[length:40%] bg-[90%_50%] bg-no-repeat'
  ),
  navigation: clsx('-mt-20 flex justify-between'),
  image: clsx('h-[12rem] w-auto select-none'),
  links: clsx('m-[5rem_2rem_0_0] flex list-none'),
  titles: clsx('absolute left-32 top-1/2 -translate-y-1/2 text-left font-[500]'),
  title: clsx('text-[6rem] text-primary'),
  subtitle: clsx('mb-12 w-1/2 text-[5rem] text-white'),
}

export default styles
