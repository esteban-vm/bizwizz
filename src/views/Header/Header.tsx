import type { FC } from 'react'
import Image from 'next/image'
import { Button, Wave } from '@/components'
import NavLink, { type NavLinkProps } from './NavLink'

export const navLinks: NavLinkProps[] = [
  { id: 'c054a01e-9103-4cd7-8a2f-c647d0d3e73b', link: '#home', text: 'Home' },
  { id: 'e67da725-6069-4902-838f-2e54b680426e', link: '#contact', text: 'Contact' },
  { id: 'fdd9fce5-598c-4029-8682-5296584eec75', link: '#about', text: 'About Us' },
]

const Header: FC = () => {
  return (
    <header
      id='home'
      className={`
        relative
        h-[100vh]
        w-full
        bg-black
        bg-[url('/images/header-img.jpg')]
        bg-[length:40%]
        bg-[90%_50%]
        bg-no-repeat
      `}
    >
      <Wave is='primary' />
      <div className='-mt-20 flex justify-between'>
        <Image
          src='/images/logo.png'
          alt='Bizwizz logo'
          width={500}
          height={500}
          className='h-[12rem] w-auto select-none'
        />
        <nav>
          <ul className='m-[5rem_2rem_0_0] flex list-none'>
            {navLinks.map((link) => (
              <NavLink key={link.id} {...link} />
            ))}
          </ul>
        </nav>
      </div>
      <div className='absolute left-20 top-60 text-left'>
        <div className='font-[500]'>
          <h1 className='text-[6rem] text-primary'>Bizwizz</h1>
          <h2 className='mb-12 w-1/2 text-[5rem] text-white'>Innovate, Transform and Lead</h2>
        </div>
        <div>
          <Button text='Get a quote today' />
        </div>
      </div>
    </header>
  )
}

export default Header
