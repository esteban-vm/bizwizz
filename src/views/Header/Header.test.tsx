import { render, cleanup, within, screen } from '@/tests'
import Header from './Header'
import navLinks from './Header.data'

describe('Header:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Header />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const title = /bizwizz/i
    const subtitle = /innovate, transform and lead/i
    const headerElem = screen.getByRole('banner')
    expect(headerElem).toBeInTheDocument()
    expect(headerElem).toHaveAccessibleName(title)
    expect(headerElem).toHaveAttribute('id', 'home')
    expect(within(headerElem).getByRole('presentation', { hidden: true })).toBeInTheDocument()
    expect(within(headerElem).getByRole('img', { name: /bizwizz logo/i })).toBeInTheDocument()
    expect(within(headerElem).getByRole('navigation')).toBeInTheDocument()
    expect(within(headerElem).getByRole('list')).toBeInTheDocument()
    expect(within(headerElem).getAllByRole('listitem')).toHaveLength(navLinks.length)
    expect(within(headerElem).getByRole('heading', { name: title, level: 1 })).toBeInTheDocument()
    expect(within(headerElem).getByRole('heading', { name: subtitle, level: 2 })).toBeInTheDocument()
    expect(within(headerElem).getByRole('button', { name: /get a quote today/i })).toBeInTheDocument()
  })
})
