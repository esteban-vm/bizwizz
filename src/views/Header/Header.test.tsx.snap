// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header: > should render correctly 1`] = `
<DocumentFragment>
  <header
    aria-labelledby="home_title"
    class="relative h-[100vh] w-full bg-black bg-[url(/images/header-img.jpg)] bg-[length:40%] bg-[90%_50%] bg-no-repeat"
    id="home"
  >
    <svg
      aria-hidden="true"
      class="wave text-primary"
      preserveAspectRatio="none"
      role="presentation"
      viewBox="0 0 500 150"
    >
      <path
        d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
        fill="currentColor"
        stroke="none"
      />
    </svg>
    <div
      class="-mt-20 flex justify-between"
    >
      <img
        alt="Bizwizz logo"
        class="h-[12rem] w-auto select-none"
        data-nimg="1"
        decoding="async"
        height="500"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flogo.png&w=1080&q=75"
        srcset="/_next/image?url=%2Fimages%2Flogo.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=1080&q=75 2x"
        style="color: transparent;"
        width="500"
      />
      <nav>
        <ul
          class="m-[5rem_2rem_0_0] flex list-none"
        >
          <li
            id="e67da725-6069-4902-838f-2e54b680426e"
          >
            <a
              class="ml-10 block select-none pb-[0.8rem]  text-[1.4rem] uppercase text-white transition-[border-b] hover:border-b-2 hover:border-b-primary"
              href="#contact"
            >
              Contact
            </a>
          </li>
          <li
            id="c054a01e-9103-4cd7-8a2f-c647d0d3e73b"
          >
            <a
              class="ml-10 block select-none pb-[0.8rem]  text-[1.4rem] uppercase text-white transition-[border-b] hover:border-b-2 hover:border-b-primary"
              href="#services"
            >
              Our services
            </a>
          </li>
          <li
            id="fdd9fce5-598c-4029-8682-5296584eec75"
          >
            <a
              class="ml-10 block select-none pb-[0.8rem]  text-[1.4rem] uppercase text-white transition-[border-b] hover:border-b-2 hover:border-b-primary"
              href="#portfolio"
            >
              Our Work
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="absolute left-20 top-60 text-left font-[500]"
    >
      <div>
        <h1
          class="text-[6rem] text-primary"
          id="home_title"
        >
          Bizwizz
        </h1>
        <h2
          class="mb-12 w-1/2 text-[5rem] text-white"
        >
          Innovate, Transform and Lead
        </h2>
      </div>
      <div>
        <button
          class="button"
          type="button"
        >
          Get a quote today
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
